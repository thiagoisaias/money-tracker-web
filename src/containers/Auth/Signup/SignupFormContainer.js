import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import { signup, clearAuthError } from "../../../store/actions/auth/auth";
import SignupForm from "../../../components/Auth/SignupForm/SignupForm";

export class SignupFormContainer extends Component {
  onSubmitData = formData => {
    const { history, isLoading, onSignup } = this.props;

    if (isLoading) {
      return;
    }

    onSignup(formData, history);
  };

  render() {
    const { onSignup, isLoading, error } = this.props;
    return (
      <SignupForm
        error={error}
        isLoading={isLoading}
        onSignup={onSignup}
        onSubmitData={this.onSubmitData}
      />
    );
  }

  componentWillUnmount() {
    const { error, onClearAuthError } = this.props;
    if (error) {
      onClearAuthError();
    }
  }
}

SignupFormContainer.propTypes = {
  error: PropTypes.string,
  onClearAuthError: PropTypes.func.isRequired,
  onSignup: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  history: PropTypes.object.isRequired
};

const mapStateToProps = state => {
  return {
    isLoading: state.auth.isLoading,
    error: state.auth.error
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onSignup: (signupData, history) => {
      dispatch(signup(signupData, history));
    },

    onClearAuthError: () => {
      dispatch(clearAuthError());
    }
  };
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(SignupFormContainer)
);
