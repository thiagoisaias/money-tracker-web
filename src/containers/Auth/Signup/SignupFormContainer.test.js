import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { SignupFormContainer } from "./SignupFormContainer";

describe("SignupFormContainer", () => {
  const mockProps = {
    error: null,
    onClearAuthError: jest.fn(),
    onSignup: jest.fn(),
    isLoading: false,
    history: {}
  };
  const component = shallow(<SignupFormContainer {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
