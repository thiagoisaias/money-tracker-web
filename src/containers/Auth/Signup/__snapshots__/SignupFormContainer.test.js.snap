// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupFormContainer renders properly 1`] = `
<Component
  error={null}
  isLoading={false}
  onSignup={[Function]}
  submitData={[Function]}
/>
`;
